src/raft.c

```diff
@@ -115,19 +115,187 @@ void entryAttachRaftReq(RedisRaftCtx *rr, raft_entry_t *entry, RaftReq *req)
 
 /* ------------------------------------ Log Execution ------------------------------------ */
 
+bool isSharding(RedisRaftCtx *rr) {
+    ShardingInfo *si = rr->sharding_info;
+
+    if (si->shard_groups_num != 1) {
+        return true;
+    }
+
+    size_t key_len;
+    ShardGroup *sg;
+
+    RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(si->shard_group_map, "^", NULL, 0);
+    RedisModule_DictNextC(iter, &key_len, (void **) &sg);
+    RedisModule_DictIteratorStop(iter);
```

Comment:
This function is in the critical path (performance-wise)  and these lines make bunch of malloc/free calls unfortunately. Probably, we can avoid this. Let's try to find something else sometime. Maybe we keep a `bool sharding` variable in ShardingInfo and update it when we apply shardgroup changes to it? Or is there any other way to do it?

Reply:
ok.  that makes sense.  looking into it.

Reply:
moved test into ShardingInfoAddShardGroup() so don't have to do test each time around.  With that said, in practice I'm unsure how big an issue this is.  This really only mattered when we had a single shard group, but change should take care of the issue.